sopel/plugins/jobs.py

```diff
@@ -51,7 +51,8 @@ class Scheduler(jobs.Scheduler):
     """
     def __init__(self, manager):
         super().__init__(manager)
-        self._jobs = tools.SopelMemoryWithDefault(list)
+        # NOTE:the annotation and type-ignore here resolves conflict with the same attribute on the base class
+        self._jobs: tools.SopelMemoryWithDefault = tools.SopelMemoryWithDefault(list)  # type: ignore[assignment]
```

Comment:
We're actually doing something surprising here, and should likely redesign the class to avoid it.

But as that's non-trivial, if others agree that we should plan to do so, it can be filed as a follow-up issue.

Reply:
I agree, I think it's a design flaw

Reply:
I agree, there is something wrong.

Also, Sopel doesn't need a JobScheduler for non-plugin things, as it can directly use the asyncio's event loop (I know, because I'm responsible for that).

Reply:
Created #2476 to follow up on this. Will mark the thread here resolved, as there's nothing inherently wrong with this band-aid fix in the mean time.